src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3444 @@
+                changeAnim = true;
+            }
+            break;
+
+        default:
+            if (animIndex != this->animIndex) {
+                changeAnim = true;
+            }
+            break;
+    }
+
+    if (changeAnim && (otherObjectSlot > OBJECT_SLOT_NONE)) {
+        gSegments[0x6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[otherObjectSlot].segment);
+        this->animIndex = animIndex;
+        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, animIndex);
+        this->unk_368 = this->skelAnime.playSpeed;
+        gSegments[0x6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[originalObjectSlot].segment);
+    }
+
+    return ret;
+}
+
+void EnAn_UpdateCollider(EnAn* this, PlayState* play) {
+    f32 temp;
+    s32 pad;
+    Vec3f sp24;
+
+    if (this->scheduleResult == ANJU_SCH_WITH_KAFEI) {
+        static Vec3f D_80B58E34 = { 0.0f, 0.0f, 8.0f };
+
+        Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &D_80B58E34, &sp24);
+        this->collider.dim.radius = 20;
+        Math_Vec3f_ToVec3s(&this->collider.dim.pos, &sp24);
+    } else {
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+    }
+
+    temp = this->actor.focus.pos.y - this->actor.world.pos.y;
+    this->collider.dim.height = TRUNCF_BINANG(temp);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
+
+typedef enum EnAnEyes {
+    /* 0 */ ENAN_EYES_OPEN,
+    /* 1 */ ENAN_EYES_HALF1,
+    /* 2 */ ENAN_EYES_CLOSED,
+    /* 3 */ ENAN_EYES_HALF2,
+    /* 4 */ ENAN_EYES_COMFORT,
+    /* 5 */ ENAN_EYES_SAD,
+    /* 6 */ ENAN_EYES_RELIEVED_CLOSED,
+    /* 7 */ ENAN_EYES_MAX
+} EnAnEyes;
+
+typedef enum EnAnMouth {
+    /* 0 */ ENAN_MOUTH_CLOSED,
+    /* 1 */ ENAN_MOUTH_HAPPY,
+    /* 2 */ ENAN_MOUTH_OPEN,
+    /* 3 */ ENAN_MOUTH_MAX
+} EnAnMouth;
+
+typedef enum EnAnFace {
+    /*  0 */ ENAN_FACE_0,
+    /*  1 */ ENAN_FACE_1,
+    /*  2 */ ENAN_FACE_2,
+    /*  3 */ ENAN_FACE_3,
+    /*  4 */ ENAN_FACE_4,
+    /*  5 */ ENAN_FACE_5,
+    /*  6 */ ENAN_FACE_6,
+    /*  7 */ ENAN_FACE_7, // RELIEVED?
+    /*  8 */ ENAN_FACE_8, // COMFORTING
+    /*  9 */ ENAN_FACE_9, // HAPPY_BLINK?
+    /* 10 */ ENAN_FACE_MAX
+} EnAnFace;
+
+void EnAn_UpdateFace(EnAn* this) {
+    static s16 sMouthIndices[ENAN_FACE_MAX] = {
+        ENAN_MOUTH_CLOSED, // ENAN_FACE_0
+        ENAN_MOUTH_OPEN,   // ENAN_FACE_1
+        ENAN_MOUTH_CLOSED, // ENAN_FACE_2
+        ENAN_MOUTH_HAPPY,  // ENAN_FACE_3
+        ENAN_MOUTH_OPEN,   // ENAN_FACE_4
+        ENAN_MOUTH_CLOSED, // ENAN_FACE_5
+        ENAN_MOUTH_HAPPY,  // ENAN_FACE_6
+        ENAN_MOUTH_OPEN,   // ENAN_FACE_7
+        ENAN_MOUTH_HAPPY,  // ENAN_FACE_8
+        ENAN_MOUTH_HAPPY,  // ENAN_FACE_9
+    };
+    s32 skipBlink = false;
+
+    if (this->stateFlags & ENAN_STATE_UPDATE_EYES) {
+        if (DECR(this->eyeTimer) == 0) {
+            switch (this->faceIndex) {
```

Comment:
default?

Reply:
iirc I tried adding default cases to all the switches and this one didn't match, but I can try again

Reply:
Indeed, it doesn't match
https://decomp.me/scratch/l25wi